graylog2-web-interface/src/components/common/SearchForm.jsx

```diff
@@ -62,21 +74,24 @@ class SearchForm extends React.Component {
     searchBsStyle: 'default',
     searchButtonLabel: 'Search',
     resetButtonLabel: 'Reset',
+    useLoadingState: false,
     loadingLabel: 'Loading...',
+    children: null,
   };
 
   state = {
+    query: this.props.query,
```

Comment:
I think we should also update the `query` in the state when the prop changes. This is specially needed if there is another way to reset the query, other than the one provided by this component.

Reply:
Aye, fixed.